### ISSUE TITLE
check for readonly variables within loops

### ISSUE THREAD

**Issue by robstoll** [koalaman/shellcheck#3089]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```bash
foo() {
  while read -r -d $'\0' file; do
    local -r someVar="..."
  done
}
```

#### Here's what shellcheck currently says:
nothing

#### Here's what I wanted or expected to see:
readonly variable within a loop is smelly as it will fail if the loop is executed more than once
